src/consensus/pbft_manager.cpp

```diff
@@ -1455,20 +1455,29 @@ std::pair<vec_blk_t, bool> PbftManager::comparePbftBlockScheduleWithDAGblocks_(P
 
 bool PbftManager::pushCertVotedPbftBlockIntoChain_(taraxa::blk_hash_t const &cert_voted_block_hash,
                                                    std::vector<Vote> const &cert_votes_for_round) {
-  if (!checkPbftBlockValid_(cert_voted_block_hash)) {
-    syncPbftChainFromPeers_(invalid_cert_voted_block, cert_voted_block_hash);
-    return false;
-  }
   auto pbft_block = pbft_chain_->getUnverifiedPbftBlock(cert_voted_block_hash);
   if (!pbft_block) {
-    LOG(log_er_) << "Can not find the cert vote block hash " << cert_voted_block_hash << " in pbft queue";
+    pbft_block = db_->getPbftCertVotedBlock(cert_voted_block_hash);
+    if (!pbft_block) {
+      LOG(log_nf_) << "Can not find the cert voted block hash " << cert_voted_block_hash
+                   << " in both pbft queue and DB";
```

Comment:
If I understood you correctly on call you said this can happen only after the node is rebooted(restarted) but I see it very often during testing and I am pretty sure the node is not restarted during those tests so there is also some other case I guess ?

Reply:
getPbftCertVotedBlock when the node cert voted a PBFT block, getUnverifiedPbftBlock when the node receive the PBFT block from peers.
When node receive enough cert votes that doesn't mean the node also receive the proposed PBFT block. In real case, if node has received 2t+1 cert votes usually the node has received the proposed PBFT block. Because node send out proposed block in step 1 and send out cert votes in step 3, there are at lease 2 lambda time (2s) between it. So in real case, when see the logs usually after node reboot (that cause unverified queue empty).
In the unit tests, since we set a very short lambda time for speed up, 2 lambda time duration is very short that caused doesn't have a significant difference to send out proposed block and cert votes. So in unit tests, node may have received enough 2t+1 cert votes but have not received block yet often. I think this is the reason